pkg/quota/evaluator/core/pods.go

```diff
@@ -144,92 +144,137 @@ func enforcePodContainerConstraints(container *v1.Container, requiredSet, missin
 }
 
 // podUsageHelper can summarize the pod quota usage based on requests and limits
-func podUsageHelper(requests v1.ResourceList, limits v1.ResourceList) v1.ResourceList {
-	result := v1.ResourceList{}
-	result[v1.ResourcePods] = resource.MustParse("1")
-	if request, found := requests[v1.ResourceCPU]; found {
-		result[v1.ResourceCPU] = request
-		result[v1.ResourceRequestsCPU] = request
+func podUsageHelper(requests api.ResourceList, limits api.ResourceList) api.ResourceList {
+	result := api.ResourceList{}
+	result[api.ResourcePods] = resource.MustParse("1")
+	if request, found := requests[api.ResourceCPU]; found {
+		result[api.ResourceCPU] = request
+		result[api.ResourceRequestsCPU] = request
 	}
-	if limit, found := limits[v1.ResourceCPU]; found {
-		result[v1.ResourceLimitsCPU] = limit
+	if limit, found := limits[api.ResourceCPU]; found {
+		result[api.ResourceLimitsCPU] = limit
 	}
-	if request, found := requests[v1.ResourceMemory]; found {
-		result[v1.ResourceMemory] = request
-		result[v1.ResourceRequestsMemory] = request
+	if request, found := requests[api.ResourceMemory]; found {
+		result[api.ResourceMemory] = request
+		result[api.ResourceRequestsMemory] = request
 	}
-	if limit, found := limits[v1.ResourceMemory]; found {
-		result[v1.ResourceLimitsMemory] = limit
+	if limit, found := limits[api.ResourceMemory]; found {
+		result[api.ResourceLimitsMemory] = limit
 	}
 	return result
 }
 
 // PodUsageFunc knows how to measure usage associated with pods
-func PodUsageFunc(object runtime.Object) v1.ResourceList {
-	pod, ok := object.(*v1.Pod)
-	if !ok {
-		return v1.ResourceList{}
-	}
-
+func PodUsageFunc(obj runtime.Object) api.ResourceList {
```

Comment:
So changing admission to be versioned is out of scope for now (for lots of reasons).

I'm primarily concerned that the function is now much harder to read than it was before.  Readability of this code is crucial.  The maintenance of the switch is the problem.  A better path may just to do:

```
func PodUsageFunc(obj runtime.Object) api.ResourceList {
  switch t := obj.(type) {
  case *api.Pod:
    if err := v1.Convert_api_Pod_to_v1_Pod(...); err != nil {
      panic("impossible conversion")
    }
  }
  // rest of method, v1 adapted
  ...
}
```

Eating errors here is terrifying because this is a security / control code path.  That would be simplest and ensures that the code is exactly the same.  Admission can be slightly slower than it is today.